export-arduino-lib.py

```diff
@@ -0,0 +1,77 @@
+import os
+import sys
+import zipfile
+import tempfile
+
+
+OUTPUT_NAME = "libsmp.zip"
+INCLUDED_DIRS = ["docs"]  # directories copied as-is with their tree
+EXTRACTED_DIRS = ["include", "src"]  # directories from which we copy all the files and put them at the root
+INCLUDED_FILES = ["COPYING", "README.md", "library.properties"]  # files in the root that we include
+EXCLUDED_FILES = [
+    ".gitignore",
+    "libsmp-private-posix.h",
+    "serial-device-avr.c",
+    "serial-device-posix.c",
+    "serial-device-win32.c"
+]  # files to be excluded globally
+CONFIGURATION_PARAMETERS = {
+    "config.h": [],
+    "libsmp-config.h": [
+        ("SMP_MESSAGE_MAX_VALUES", 8, "The maximum number of arguments in a message"),
+        ("SMP_SERIAL_FRAME_MAX_FRAME_SIZE", 128, "The maximum frame size in bytes")
+    ]
+}
+
+if len(sys.argv) > 1:
+    OUTPUT_NAME = sys.argv[1]  # If provided, we use the filename given by the user
+
+
+def add_dir_to_zip(zip_file, path, keep_path):
+    if os.path.isdir(path):
+        for root, dirs, files in os.walk(path):
+            for f in files:
+                if f not in EXCLUDED_FILES:
+                    if keep_path:
+                        zip_file.write(os.path.join(root, f))
+                    else:
+                        zip_file.write(os.path.join(root, f), f)
+    else:
+        print(d, ": no such directory")
+
+
+def generate_config_file(params):
+    cf = tempfile.NamedTemporaryFile(mode='w', delete=False)
+    cf.write("#pragma once\n\n")
+    for param in params:
+        print(param[0] + " [default=" + str(param[1]) + "]  " + param[2])
+        try:
+            value = int(input("> "))  # We try to get the value from the user
+        except ValueError:
+            value = int(param[1])  # If the user doesn't provide a value, we use the default
+        print(param[0], "=", value)
+        print()
+        cf.write("/* " + param[2] + " */\n")
+        cf.write("#define " + param[0] + " " + str(value) + "\n\n")
+    cf.close()
+    return cf.name
+
+
+output_file = zipfile.PyZipFile(OUTPUT_NAME, 'w')
+
+for file in INCLUDED_FILES:
```

Comment:
we should build path to our files, based on the script location. Otherwise, if the script is not ran from the good directory, it will fail.